Solutions/Corvus.Testing.AzureFunctions/Corvus/Testing/AzureFunctions/FunctionsController.cs

```diff
@@ -32,6 +35,26 @@ public sealed class FunctionsController
         private readonly List<FunctionOutputBufferHandler> output = new List<FunctionOutputBufferHandler>();
         private readonly object sync = new object();
 
+        private readonly ILogger logger;
+        private readonly AsyncLocal<IDisposable?> functionLogScope = new AsyncLocal<IDisposable?>();
```

Comment:
I don't follow the need for this to be `AsyncLocal`. This gets set in `StartFunctionsInstance`, so the implication of this change is that you expect a single instance of `FunctionsController` to see multiple calls to `StartFunctionsInstance` in flight concurrently (with eventual corresponding calls to `TeardownFunctions` coming on on the same async causality).

My memory is a bit hazy on this but I thought that a single `FunctionsController` would normally only be expected to be responsible for a single hosted function instance?

Reply:
Hi @alastairs - I'd like to approve this PR, but this part looks peculiar to me and I'd like to understand it first. Just pinging again in case you didn't see the earlier reply.

Reply:
@idg10 IIRC some of the tests failed with an async-y error message, which this fixed. To be frank, it was a bit of a blind stab in the dark, so I'll take another look.

Reply:
@idg10 I'm sorry for the delay resolving this. I've removed the `AsyncLocal` wrapper to expedite the PR. From memory, the failure occurred in our library that integrates Corvus.Testing into NUnit. If that is still the case after the PR is merged, then I'll do more debugging to find the source of the issue and either fix it in our library or raise another issue here. 

That said, I've just now seen the comment I left in f2beb11, which indicates it might be specific to using the Microsoft `ConsoleLogger` and a NullReferenceException thrown there, so there's a couple of avenues to investigate when it comes to it.